nodestream_plugin_neptune/database_connector.py

```diff
@@ -1,17 +1,88 @@
-from nodestream.databases import TypeRetriever, DatabaseConnector
-from nodestream.databases.query_executor import QueryExecutor
+from nodestream.databases.copy import TypeRetriever
+from nodestream.databases.database_connector import QueryExecutor, DatabaseConnector
+from .ingest_query_builder import NeptuneDBIngestQueryBuilder
 
-from .query_executor import NeptuneQueryExecutor
-from .type_retriever import NeptuneTypeRetriever
 
+class NeptuneDatabaseConnector(DatabaseConnector, alias="neptune_db"):
+    @classmethod
+    def from_file_data(
+        cls,
+        host: str,
+        region: str,
+        **kwargs
+    ):
+        # Make this use boto3
+        return cls(
+            host=host,
+            region=region,
+            async_partitions=kwargs.get("async_partitions"),
+            ingest_query_builder=NeptuneDBIngestQueryBuilder()
+        )
 
-class NeptuneDatabaseConnector(DatabaseConnector, alias="neptune"):
-    """A Connector for a Neptune Graph Databases.
+    def __init__(
+        self,
+        region,
+        host,
+        async_partitions,
+        ingest_query_builder: NeptuneDBIngestQueryBuilder
+    ) -> None:
+        self.host = host
+        self.region = region
+        self.ingest_query_builder = ingest_query_builder
+        self.async_partitions = async_partitions
 
-    This class is responsible for creating the various components needed for
-    nodestream to interact with a Neptune database. It is also responsible
-    for providing the configuration options for the Neo4j database.
-    """
+    def make_query_executor(self) -> QueryExecutor:
+        from .query_executor import NeptuneDBQueryExecutor
 
-    # TODO: Implement this.
-    pass
+        return NeptuneDBQueryExecutor(
+            host=self.host,
+            region=self.region,
+            ingest_query_builder=self.ingest_query_builder,
+            async_partitions=self.async_partitions
+        )
+
+    def make_type_retriever(self) -> TypeRetriever:
+        from .type_retriever import NeptuneDBTypeRetriever
+
+        return NeptuneDBTypeRetriever(self)
+
+    def make_migrator(self) -> TypeRetriever:
+        raise NotImplementedError
```

Comment:
This could be left as an issue for now. As far as I currently understand the migrator is mostly not applicable to Neptune currently as Neptune does not support custom indices or constraints. We should have a discussion about what a Neptune migrator might look like when added.